modules/tinymce/src/plugins/table/test/ts/browser/settings/TableToolbarTest.ts

```diff
@@ -50,4 +52,55 @@ describe('browser.tinymce.plugins.table.TableToolbarTest', () => {
     await TinyUiActions.pWaitForUi(editor, '.tox-pop .tox-tbtn[title="Paste row before"]:not(.tox-tbtn--disabled)');
     McEditor.remove(editor);
   });
+
+  context('Ensure table buttons are enabled when appropriate', () => {
+    it('TINY-7737: When the whole selection is in the table', async () => {
+      const editor = await pCreateEditor('tablecaption');
+      editor.focus();
+      editor.setContent(tableWithAdditionalHtml);
+      await TinyUiActions.pWaitForUi(editor, '.tox-tbtn--disabled[title="Table caption"]');
+      TinySelections.setSelection(editor, [ 1, 0, 0, 0 ], 0, [ 1, 0, 0, 0 ], 1);
+      // Wait for a while to allow the toolbar a chance to render
+      await Waiter.pWait(100);
```

Comment:
This wait should not be here and in the other tests, as the toolbar will always be rendered.